api/types.go

```diff
@@ -609,6 +609,59 @@ func (pins PinSerial) ToPin() Pin {
 	}
 }
 
+// AddedOutput carries information for displaying the standard ipfs output
+// indicating a node of a file has been added.
+type AddedOutput struct {
+	Error
+	Name  string
+	Hash  string `json:",omitempty"`
+	Bytes int64  `json:",omitempty"`
+	Size  string `json:",omitempty"`
+}
+
+// NodeSerial encodes info necessary to add an IPFS Block. This includes the
+// cid, data, and size
+type NodeSerial struct {
+	Cid  string
+	Data []byte
+	Size uint64
+}
+
+// ShardNodeSerial encodes the info necessary to add an IPFS Block to a sharded
+// file.  This includes the serialized node and a sharding ID.
+type ShardNodeSerial struct {
+	Cid  string
+	Data []byte
+	Size uint64
+	ID   string
+}
+
+// BlockWithFormat specifies a block of data and how it should be
+// formatted.  An empty format string will be interpreted as "v0".
+type BlockWithFormat struct {
```

Comment:
Is this mergeable with NodeSerial? Meaning, that we could use NodeSerial for passing a block and its data around?

I'm even wondering if worth merging with `ShardNodeSerial` too.

Reply:
For sure, if you're ok with having unrelated zero valued fields along for the ride.  It's a trade off between lots of specific types perfectly tuned to match function call parameters and fewer types that are used in a bunch of different ways.  I think I've been doing the former lately when defining api types but I'm happy to move towards the latter.

Reply:
I think, since we only use this to pass data internally, and this data is essentially the same (a block slice with metadata that gets filled in as it goes down the pipeline), we can use the same type for everything. It's much simpler to see that the same type leaving the rest api is the one arriving to the ipfs connector. Essentially it hasn't been transformed, just carries more info in it.

Reply:
Got it.  Do you think ShardNodeSerial should also be merged into NodeSerial?

Reply:
Yes